middleware_oauth.go

```diff
@@ -3,17 +3,18 @@ package main
 import (
 	"encoding/json"
 
+	"errors"
+	"net/http"
```

Comment:
Yeah! That is exactly what I'm talking about.
> Imports are organized in groups, with blank lines between them. The standard library packages are in the first group.

And you have an empty line between "encoding/json" and "errors". They are both from standard library